File: Lab1.c
// Lab 1: Implementation of Stack as an ADT
#include <stdio.h>
#include <stdlib.h>
#define MAX 5
#define true 1
#define false 0

struct Stack
{
  int TOP;
  int data[MAX];
};

int isFull(struct Stack S)
{
  return (S.TOP == MAX - 1) ? true : false;
}

int isEmpty(struct Stack S)
{
  return (S.TOP == -1) ? true : false;
}

void push(struct Stack *S, int element)
{
  if (isFull(*S))
    printf("STACK OVERFLOW\n");
  else
  {
    S->TOP += 1;
    S->data[S->TOP] = element;
    printf("SUCCESS\n");
  }
}

int pop(struct Stack *S)
{
  if (isEmpty(*S))
  {
    printf("STACK UNDERFLOW");
    return -1;
  }
  else
  {
    return S->data[S->TOP--];
  }
}

int main()
{
  int choice, element;
  struct Stack AnyThing = {-1};
  // AnyThing.Top = -1
  do
  {
    printf("1. PUSH\n2. POP\n3. EXIT\n");
    printf("Choice? ");
    scanf("%d", &choice);
    switch (choice)
    {
    case 1:
      printf("Enter Element: ");
      scanf("%d", &element);
      push(&AnyThing, element);
      break;

    case 2:
      printf("%d WAS POPPED\n", pop(&AnyThing));
      break;

    case 3:
      printf("BYE BYE\n");
      break;

    default:
      break;
    }
  } while (choice != 3);

  return 0;
}

File: Lab1.py
def push(stack, element):
  stack.append(element)
  print("SUCCESS")
  
def popp(stack):
  if len(stack) == 0:
    return "Stack Underflow"
  else:
    data = stack.pop()
    return data
  
flag = True
stack = list()
while(flag):
  print("1. PUSH")
  print("2. POP")
  print("3. EXIT")
  choice = int(input("Enter yout choice: "))
  if choice == 1:
    element = input("Enter Element")
    push(stack, element)
  elif choice ==  2:
    element = popp(stack)
    print("Popped Value = ", element)
  else:
    flag = False